packages/server/lib/controllers/connection.controller.ts

```diff
@@ -527,6 +528,21 @@ class ConnectionController {
                 return;
             }
 
+            const account = await accountService.getAccountById(accountId);
+
+            if (!account) {
+                errorManager.errRes(res, 'unknown_account');
+                return;
+            }
+
+            if (account.is_capped && provider_config_key) {
+                const isCapped = await connectionCreationStartCapCheckHook({ providerConfigKey: provider_config_key, environmentId });
+                if (isCapped) {
+                    errorManager.errRes(res, 'resource_capped');
+                    return;
+                }
+            }
```

Comment:
this is already checked in the middleware right?
tbf I would only keep this check here since it's really specific to this use case

Reply:
This method doesn't go through the middleware, only the calls that are attempting to creating a connection using the public auth key go through that middleware.

Reply:
yeah okay, got confused since we have 3 apis 👌🏻